src/util.c

```diff
@@ -0,0 +1,41 @@
+#define _GNU_SOURCE
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+static __thread bool nm_err_state                       = false;
+static __thread char nm_err_buf[2048]                   = {0};
+static __thread char nm_err_buf_tmp[sizeof(nm_err_buf)] = {0}; // in case the format string overlaps
+
+const char *nm_err() {
+    if (nm_err_state) {
+        nm_err_state = false;
+        return nm_err_buf;
+    }
+    return NULL;
+}
+
+const char *nm_err_peek() {
+    if (nm_err_state)
+        return nm_err_buf;
+    return NULL;
+}
+
+bool nm_err_set(const char *fmt, ...) {
+    va_list a;
+    va_start(a, fmt);
+    if ((nm_err_state = !!fmt)) {
```

Comment:
Coercing `NULL`/`nullptr` to a boolean makes `false`? Cool!

Reply:
I guess the `va_` macros could also be moved solely inside the `*nprintf` branch?

Reply:
Done.